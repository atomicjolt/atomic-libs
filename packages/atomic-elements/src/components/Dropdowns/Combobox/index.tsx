export { ComboBox } from "./ComboBox.component";
export type { ComboBoxProps } from "./ComboBox.component";
