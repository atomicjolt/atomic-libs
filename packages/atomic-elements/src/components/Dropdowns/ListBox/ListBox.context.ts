import { createContext } from "react";
import { ListState } from "react-stately";
import { createComponentContext } from "@utils/index";
import { ListBoxProps } from "./ListBox.types";

export const ListStateContext = createContext<ListState<any> | null>(null);
export const ListBoxContext = createComponentContext<ListBoxProps<any>>();
