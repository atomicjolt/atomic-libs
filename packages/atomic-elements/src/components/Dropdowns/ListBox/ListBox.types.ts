import { AriaListBoxProps } from "@react-aria/listbox";
import { ExtendedSize, RenderStyleProps } from "../../../types";

export interface ListBoxProps<T>
  extends AriaListBoxProps<T>,
    RenderStyleProps<never> {
  /** The size of the listbox */
  size?: ExtendedSize;
}
