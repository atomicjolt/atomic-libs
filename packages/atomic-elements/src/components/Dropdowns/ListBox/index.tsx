export { ListBox, InternalListBox } from "./ListBox.component";
export { ListBoxContext, ListStateContext } from "./ListBox.context";
export type { ListBoxProps } from "./ListBox.types";
