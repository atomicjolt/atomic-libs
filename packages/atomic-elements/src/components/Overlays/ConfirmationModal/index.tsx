export { ConfirmationModal } from "./ConfirmationModal.component";
export type { ConfirmationModalProps } from "./ConfirmationModal.component";
