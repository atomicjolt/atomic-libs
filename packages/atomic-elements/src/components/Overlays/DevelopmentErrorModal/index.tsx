export {
  DevelopmentErrorModal,
  DevelopmentErrorBoundary,
} from "./DevelopmentErrorModal.component";
export type { DevelopmentErrorModalProps } from "./DevelopmentErrorModal.component";
