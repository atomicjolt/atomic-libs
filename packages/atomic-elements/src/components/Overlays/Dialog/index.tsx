export { Dialog } from "./Dialog.component";
export type { DialogProps } from "./Dialog.component";
