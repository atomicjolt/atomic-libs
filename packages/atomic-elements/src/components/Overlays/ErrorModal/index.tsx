export { ErrorModal } from "./ErrorModal.component";
export type { ErrorModalProps } from "./ErrorModal.component";
