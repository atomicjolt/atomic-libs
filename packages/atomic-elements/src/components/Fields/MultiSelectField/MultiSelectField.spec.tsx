import { describe, test, expect } from "vitest";
import { render } from "@testing-library/react";
import { Popover } from "@components/Overlays/Popover";
import { ListBox } from "@components/Dropdowns/ListBox";
import { MultiSelectField } from ".";
import { Item } from "@components/Collection";

describe("MultiSelectField", () => {
  describe("matches snapshots", () => {
    test("when closed", () => {
      const result = render(
        <MultiSelectField>
          <MultiSelectField.Button>Open up</MultiSelectField.Button>
          <Popover>
            <ListBox>
              <Item>Item 1</Item>
              <Item>Item 2</Item>
              <Item>Item 3</Item>
            </ListBox>
          </Popover>
        </MultiSelectField>
      );
      expect(result.asFragment()).toMatchSnapshot();
    });

    test("when open", () => {
      const result = render(
        <MultiSelectField isOpen>
          <MultiSelectField.Button>Open up</MultiSelectField.Button>
          <Popover>
            <ListBox>
              <Item>Item 1</Item>
              <Item>Item 2</Item>
              <Item>Item 3</Item>
            </ListBox>
          </Popover>
        </MultiSelectField>
      );
      expect(result.asFragment()).toMatchSnapshot();
    });
  });
});
