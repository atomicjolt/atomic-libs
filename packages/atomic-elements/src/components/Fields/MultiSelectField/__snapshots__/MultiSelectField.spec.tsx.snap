// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MultiSelectField > matches snapshots > when closed 1`] = `
<DocumentFragment>
  <template
    data-react-aria-hidden="true"
  />
  <div
    class="sc-gpHbIA bhSUCw aje-multiselect"
  >
    <button
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="react-aria-:r7: react-aria-:r3:"
      class="sc-cMqPwU fLpmHx aje-btn aje-btn--dropdown is-medium"
      id="react-aria-:r2:"
      tabindex="0"
      type="button"
    >
      Open up
    </button>
  </div>
</DocumentFragment>
`;

exports[`MultiSelectField > matches snapshots > when open 1`] = `
<DocumentFragment>
  <template
    data-react-aria-hidden="true"
  />
  <div
    class="sc-gpHbIA bhSUCw aje-multiselect"
  >
    <button
      aria-controls="react-aria-:r9:"
      aria-expanded="true"
      aria-haspopup="listbox"
      aria-labelledby="react-aria-:rf: react-aria-:rb:"
      class="sc-cMqPwU fLpmHx aje-btn aje-btn--dropdown is-medium"
      data-pressed="true"
      id="react-aria-:ra:"
      tabindex="0"
      type="button"
    >
      Open up
    </button>
  </div>
</DocumentFragment>
`;
